aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -799,4 +800,241 @@ Tensor sspaddmm(const Tensor& self, const Tensor& mat1, const Tensor& mat2,
   return result;
 }
 
+// --------------------------------------------------------------------
+// sparse.sum()
+// --------------------------------------------------------------------
+Tensor _sparse_sum(const SparseTensor& input) {
+  return input.values().sum();
+}
+
+Tensor _sparse_sum(const SparseTensor& input, ScalarType dtype) {
+  return at::_sparse_sum(input.to(dtype));
+}
+
+Tensor _sparse_sum(const SparseTensor& input, IntList dims_to_sum) {
+  return at::_sparse_sum(input, dims_to_sum, false);
+}
+
+Tensor _sparse_sum(const SparseTensor& input, IntList dims_to_sum, ScalarType dtype) {
+  return at::_sparse_sum(input.to(dtype), dims_to_sum, false);
+}
+
+Tensor _sparse_sum(const SparseTensor& input, IntList dims_to_sum, bool keepdim) {
+
+  AT_CHECK(input.is_coalesced(), "To support autograd, input SparseTensor has to be coalesced.");
```

Comment:
why can't you do `coalesce` for users here?

Reply:
commented above